### PR TITLE
Update client and stream server to allow redactions from REACT user.

### PR DESCRIPTION
**Pull request by clemire**
Update the react permission implementation to ensure that REACT users can redact their reactions.

### CODE REVIEW

File: core/node/rules/can_add_event.go [hunk 1]
```diff
@@ -204,7 +204,7 @@ func (params *aeParams) canAddChannelPayload(payload *StreamEvent_ChannelPayload
 	case *ChannelPayload_Redaction_:
 		return aeBuilder().
 			check(params.creatorIsMember).
-			requireChainAuth(params.redactChannelMessageEntitlements)
```
Comment (texuf) [inline core/node/rules/can_add_event.go:207]: a sendChannelMessage_Redaction does not result in a ChannelPayload_Redaction_, it's a ChannelPayload_Message. `redactMessage` results in a ChannelPayload_Redaction_, does this change your assumptions about this PR?

Comment (clemire) [reply core/node/rules/can_add_event.go:207]: Ah, I made a mistake here. My original PR sent a redaction and I fixed it to send a message with a redaction payload and just added a bug.

Comment (clemire) [reply core/node/rules/can_add_event.go]: Thank you for the edifying comment!

Comment (clemire) [reply core/node/rules/can_add_event.go]: I believe the channel message with redaction payload type is the right one here... We discussed it a while ago - this is the self-redaction option, does not require special permissions. and redactMessage is an admin-only client method.

**Review by texuf — APPROVED** [review]

**Pull request merged**